auslib/db.py

```diff
@@ -928,16 +929,20 @@ def __init__(self, db, dialect, metadata, baseTable, conditions=("time", "uptake
             # See https://github.com/zzzeek/sqlalchemy/blob/rel_0_7/lib/sqlalchemy/schema.py#L781
             # for background.
             newcol.key = newcol.name = "base_%s" % col.name
-            # 2) Primary keys on the base table are normal columns here,
-            # and are allowed to be null (because we support adding new rows
-            # to tables with autoincrementing keys via scheduled changes).
-            # The base table's data version may also be null for the same reason.
-            if col.primary_key or newcol.name == "base_data_version":
+            # 2) All non PK columns from baseTable should be nullable. Base table PK columns
+            # which are autoincrement should be nullable in the Scheduled Changes version of
+            # the column. Base table PK columns which are not autoincrement should not be nullable.
```

Comment:
A Scheduled Change needs to know when baseTable row it will be operating on, or inserting.

For baseTable PK columns which are autoincrement integers, this will be determined automatically when the Scheduled Change insert is enacted. Technically, we shouldn't allow NULLs for these columns if change_type is update or delete (otherwise we don't know which baseTable row to update or delete) - this gets enforced in validate(): https://github.com/mozilla/balrog/blob/897c39144caafaa41c53bfdd95c70f71e66b4b03/auslib/db.py#L1017.

For baseTable PK columns which aren't autoincrement, or aren't integers, we *always* need a value, even for inserts because there's no way to create it at enact time.

Does that help?